### ISSUE TITLE
Need dependencies argument for cc.links()

### ISSUE THREAD

**Issue by tp-m** [mesonbuild/meson#367]
Unfortunately some libraries don't ship with pkg-config files OR they allow dropping/disabling of features whilst still shipping full headers and there's no way to tell from the headers what features are available. Sometimes these libs are old and practically unmaintained, so fixing it is not always an option.

For those cases we need to be able to check if a lib actually contains a certain symbol.

Neither cc.has_function() nor cc.compiles() handle this case. We basically need the equivalent of AC_CHECK_LIB().

Case in point is libvpx, but there's also a bunch of other/old libs that need this.

**Comment by jpakkane** [comment]
I trust you have tried links(). What extra functionality on top of that would this require?

**Comment by tp-m** [comment]
Ah, sorry, I have not tried .links() yet. I didn't see it listed on https://github.com/mesonbuild/meson/wiki/Reference-manual under 'compiler object'. Will try and report back, thanks!

**Comment by tp-m** [comment]
Ok, so cc.links() works in principle, but I think it's lacking some features, namely the possibility to do something like:

```
vpx_dep = dependency('vpx', version : '>= 1.4.0')
if cc.links(link_code, dependencies : vpx_dep)
  ...
endif
```

to make sure the right include paths are used and the right linker flags (the lib might be installed in a different prefix).

Same for being able to pass an object returned from find_library() I suppose.

**Comment by nirbheek** [comment]
Fixed via 13e91ab4996ffd3e3ebc24e7a69ccba64e89b4f5
